conans/client/source.py

```diff
@@ -72,7 +74,8 @@ def remove_source(raise_error=True):
         try:
             with tools.environment_append(conan_file.env):
                 with conanfile_exception_formatter(str(conan_file), "source"):
-                    conan_file.source()
+                    with block_build_and_package_folders():
```

Comment:
If the user tries to access build_folder or package_folder will raise an error explaining.